cmd/builder/internal/builder/config.go

```diff
@@ -89,15 +90,17 @@ func NewDefaultConfig() Config {
 // Validate checks whether the current configuration is valid
 func (c *Config) Validate() error {
 	// #nosec G204
-	if _, err := exec.Command(c.Distribution.Go, "env").CombinedOutput(); err != nil {
-		path, err := exec.LookPath("go")
-		if err != nil {
-			return ErrGoNotFound
+	if !(c.SkipCompilation && c.SkipGetModules) {
```

Comment:
This change seems unrelated to the original intent of your PR. How about splitting it into another one?

Also, a global negative condition not the most readable way to write this IMHO.
I would write this `if !c.SkipCompilation || !c.SkipGetModules`.

Reply:
Good point. I will move it to a separate PR and create a new one once this PR is merged.